File: android/android/src/de/benibela/videlibri/BookListFragment.java
package de.benibela.videlibri;

import android.widget.*;

public class BookListFragment extends VideLibriFakeFragment{
    private ListView bookListView;

    BookListFragment(final BookListActivityOld activity){
        super(activity);

        bookListView = (ListView) findViewById(R.id.booklistview);
        if (bookListView != null) activity.registerForContextMenu(bookListView);
    }


    BookOverviewAdapter getAdapter(){
        return (BookOverviewAdapter) bookListView.getAdapter();
    }

    String exportShare(boolean html){
        BookOverviewAdapter adapter = getAdapter();
        return adapter.exportShare(html);
    }

}

File: android/android/src/de/benibela/videlibri/BookListFragment.kt
package de.benibela.videlibri

import de.benibela.multilevellistview.ClickableRecyclerView

class BookListFragment internal constructor(activity: BookListActivity) : VideLibriFakeFragment(activity) {
    val listview: ClickableRecyclerView = activity.findViewById(R.id.bookOverviewRecyclerView)
    internal var adapter
        get() = listview.adapter as? BookOverviewAdapter
        set(value) { listview.adapter = value }

    fun exportShare(html: Boolean): String {
        val adapter = adapter
        return adapter?.exportShare(html) ?: ""
    }
}